packages/scoped-elements/README.md

```diff
@@ -0,0 +1,276 @@
+2. [no-scope](https://open-wc.org/scoped-elements/demo/no-scoping/) [[code](https://github.com/open-wc/open-wc/tree/master/packages/scoped-elements/demo/no-scope)] Feature A version 1.x and 2.x are imported via self registering entry points which leads to the following error message as "feature-a" tries to register multiple times
+
+   ```
+   Uncaught DOMException: Failed to execute 'define' on 'CustomElementRegistry': the name "feature-a" has already been used with this registry
+     at [...]/node_modules/page-b/node_modules/feature-a/feature-a.js:3:16
+   ```
+
+3. [with-scope](https://open-wc.org/scoped-elements/demo/with-scoping/) [[code](https://github.com/open-wc/open-wc/tree/master/packages/scoped-elements/demo/with-scope)] Successfully "fixing" the demo by using `createScopedHtml` on Page A/B.
+
+## How it works
+
+`createScopedHtml` is going to define the custom element classes and creates a
+unique scoped tag for each of them if needed. In case those classes were
+previously defined then it's going to return the previous defined tag.
+
+Then, the `html` function provided by the `createScopedHtml` method is going to
+transform the template literal into another one replacing the tags used by the
+user by the ones defined by the custom elements. Finally, the transformed
+template literal is going to be processed by `lit-html`.
+
+`<my-button>${this.text}</my-button>` --> `<my-button-3>${this.text}</my-button-3>`
+
+## How to use it
+
+1. Import the `createScopedHtml` from `@open-wc/scoped-elements`.
+
+   ```js
+   import { createScopedHtml } from '@open-wc/scoped-elements';
+   ```
+
+2. Import the classes of the components you want to use.
+
+   ```js
+   import MyButton from './MyButton.js';
+   import MyPanel from './MyPanel.js';
+   ```
+
+3. Create the `html` function defining the tags you want to use for the
+   components.
+
+   ```js
+   const html = createScopedHtml({
+     'my-button': MyButton,
+     'my-panel': MyPanel,
+   });
+   ```
+
+4. Use your components in your html.
+
+   ```js
+   render() {
+     return html`
+       <my-panel class="panel">
+         <my-button>${this.text}</my-button>
+       </my-panel}>
+     `;
+   }
+   ```
+
+### Complete example
+
+```js
+import { css, LitElement } from 'lit-element';
+import { createScopedHtml } from '@open-wc/scoped-elements';
+import MyButton from './MyButton.js';
+import MyPanel from './MyPanel.js';
+
+const html = createScopedHtml({
+  'my-button': MyButton, // <-- bind the element with a tag in your html
+  'my-panel': MyPanel,
+});
+
+export default class MyElement extends LitElement {
+  static get styles() {
+    return css`
+      .panel {
+        padding: 10px;
+        background-color: grey;
+      }
+    `;
+  }
+
+  static get properties() {
+    return {
+      text: String,
+    };
+  }
+
+  render() {
+    return html`
+      <my-panel class="panel">
+        <my-button>${this.text}</my-button>
+      </my-panel}>
+    `;
+  }
+}
+```
+
+## Limitations
```

Comment:
Although it might be expected, I would add as a limitation: 
- that all dependencies should be NPM dependencies
- that all references should use node resolution

Reply:
hmm I do not think it matters where you get your dependencies from?
as soon as it is double-loaded (be it nested npm or same file via different imports or a micro architecture with some double loading)

Reply:
I think it depends. 🙈 You are right by saying it should be double-loaded (or multi-loaded for that matter). If you use Bower to package the components, you won't be able to multi-load. But it's true that within a project it doesn't matter. 😃   
When I mentioned "dependencies", I was talking about packaging strategy, especially valid for the multi teams example.